x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -52,6 +52,10 @@ export const searchAfterAndBulkCreate = async ({
   if (someResult.hits.hits.length === 0) {
     return true;
   }
+  const { index, from, to } = ruleParams;
+  if (index == null) {
+    throw new Error('Attempted to bulk create signals, but rule had no indexPattern');
```

Comment:
We might want to use your `BadRequest` here as well for consistency? This isn't really a REST request though, so I am on the fence.

I would tune the words a bit to remove the camel case word of `indexPattern` and add as much useful information as possible in case this ever gets hit we can trouble shoot better:

```ts
    throw new Error(
      `Attempted to bulk create signals, but rule id: ${id}, name: ${name}, signals index: ${signalsIndex} has no index pattern`
    );
```

That way we will know which rule id, name, and which space on a customer site to have them delete or remove or investigate if this ever does get touched.